packages/ts/src/components/nested-donut/index.ts

```diff
@@ -0,0 +1,162 @@
+import { Selection } from 'd3-selection'
+import { arc } from 'd3-shape'
+import { hierarchy, partition } from 'd3-hierarchy'
+import { scaleLinear, ScaleLinear } from 'd3-scale'
+import { max, group, sum, min } from 'd3-array'
+
+// Core
+import { ComponentCore } from 'core/component'
+import { SeriesDataModel } from 'data-models/series'
+
+// Types
+import { VerticalAlign } from 'types/text'
+
+// Utils
+import { getColor, getHexValue } from 'utils/color'
+import { getString, getValue, isNumber } from 'utils/data'
+import { smartTransition } from 'utils/d3'
+import { wrapSVGText } from 'utils/text'
+
+// Styles
+import * as s from './style'
+
+// Config
+import { NestedDonutConfig, NestedDonutConfigInterface } from './config'
+import { createArc, updateArc, removeArc } from './modules/arc'
+import { NestedDonutDirection, NestedDonutLayerSettings, NestedDonutSegment, defaultLayerSettings } from './types'
+
+export class NestedDonut<Datum> extends ComponentCore<
+Datum[],
+NestedDonutConfig<Datum>,
+NestedDonutConfigInterface<Datum>
+> {
+  static selectors = s
+  config: NestedDonutConfig<Datum> = new NestedDonutConfig()
+  datamodel: SeriesDataModel<Datum> = new SeriesDataModel()
+
+  arcBackground: Selection<SVGPathElement, unknown, SVGGElement, unknown>
+  arcGroup: Selection<SVGGElement, unknown, SVGGElement, unknown>
+  centralLabel: Selection<SVGTextElement, unknown, SVGGElement, unknown>
+  centralSubLabel: Selection<SVGTextElement, unknown, SVGGElement, unknown>
+
+  arcGen = arc<Partial<NestedDonutSegment<Datum>>>()
+  radiusScale: ScaleLinear<number, number> = scaleLinear()
+  colorScale: ScaleLinear<string, string> = scaleLinear()
+
+  events = { }
+
+  constructor (config?: NestedDonutConfigInterface<Datum>) {
+    super()
+    if (config) this.config.init(config)
+    this.arcBackground = this.g.append('path')
+    this.arcGroup = this.g.append('g')
+    this.centralLabel = this.g.append('text')
+      .attr('class', s.centralLabel)
+    this.centralSubLabel = this.g.append('text')
+      .attr('class', s.centralSubLabel)
+  }
+
+  private getHierarchyData (): NestedDonutSegment<Datum>[] {
+    const { config, datamodel } = this
+
+    const layerAccessors = config.layers.map(layerAccessor => (d: Datum) => getString(d, layerAccessor))
```

Comment:
@reb-dev If there's a way to provide the datum index here, I think we should do it: `getString(d, layerAccessor, i)`